### PR TITLE
Disable EOF CALL opcodes in legacy

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

When the EOF call opcodes are added to the EVM, ensure legacy cannot it.

### Thanks for sending a pull request! Have you done the following?

- [ ] Checked out our [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)?
- [ ] Considered documentation and added the `doc-change-required` label to this PR [if updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).
- [ ] Considered the changelog and included an [update if required](https://wiki.hyperledger.org/display/BESU/Changelog).
- [ ] For database changes (e.g. KeyValueSegmentIdentifier) considered compatibility and performed forwards and backwards compatibility tests

### Locally, you can run these tests to catch failures early:

- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`
- [ ] integration tests: `./gradlew integrationTest`
- [ ] reference tests: `./gradlew ethereum:referenceTests:referenceTests`

### CODE REVIEW

File: evm/src/main/java/org/hyperledger/besu/evm/operation/AbstractExtCallOperation.java [hunk 1]
```diff
@@ -95,6 +95,11 @@ public long gasAvailableForChildCall(final MessageFrame frame) {
 
   @Override
   public OperationResult execute(final MessageFrame frame, final EVM evm) {
+    Code callingCode = frame.getCode();
+    if (callingCode.getEofVersion() == 0) {
+      return InvalidOperation.INVALID_RESULT;
```
Comment (jframe) [inline evm/src/main/java/org/hyperledger/besu/evm/operation/AbstractExtCallOperation.java:100]: Would be good to add some tests showing that an INVALID_RESULT is returned for legacy with EXTCALL, EXTDELEGATECALL and EXTSTATICCALL

Comment (shemnon) [reply evm/src/main/java/org/hyperledger/besu/evm/operation/AbstractExtCallOperation.java:100]: https://github.com/ethereum/execution-spec-tests/pull/768/files

Tests all legacy-invalid opcodes, not just EXT*CREATE.

Comment (garyschulte) [reply evm/src/main/java/org/hyperledger/besu/evm/operation/AbstractExtCallOperation.java:100]: LGTM, we can pull in the latest execution-spec tests when they get published :+1:

**Review by garyschulte — APPROVED** [review]
🚢

File: evm/src/main/java/org/hyperledger/besu/evm/operation/AbstractExtCallOperation.java (hunk 1 above)
Comment (shemnon) [reply evm/src/main/java/org/hyperledger/besu/evm/operation/AbstractExtCallOperation.java:100]: https://github.com/hyperledger/besu/pull/7544/commits/f19e24279753b2a99af03b5f4f645961b632adca also should address @jframe's concern

**Pull request merged**